docs/source/releases/latest/575-cli-class-factory.yaml

```diff
@@ -0,0 +1,27 @@
+Refactoring Updates:
+- description: |
+    *From GEOIPS#576: 2024-05-11, CLI: Reduce number of command layers where possible and appropriate*
+    *From GEOIPS#573: 2024-05-11, Use class factories to generate subcommands for commands like ``geoips get family`` and ``geoips get interface``*
+
+    While this PR doesn't fix all of the problems addressed in the issues above, it is our
+    initial attempt at creating a class factory for certain CLI commands to reduce the
+    verbage needed to execute those commands. Spefically in this branch, we've addressed the
+    command class ``GeoipsListSingleInterface``. We now use the aforementioned class as a
+    base class to build ``GeoipsListSingleInterface<interface_name>`` classes at runtime.
+
+    By doing so, we can now run ``geoips list <interface_name>`` rather than
+    ``geoips list interface <interface_name>``. This improves the readability of this
+    command and makes it much easier to type via the CLI. We expect in future PRs to address
+    similar commands, such as ``geoips get family <family_name>`` and
+    ``geoips get plugin <interface_name> <plugin_name>`` for the reasons mentioned
+    previously.
```

Comment:
@evrose54 @jsolbrig Wait, what happens if you have a plugin named the same thing in 2 different interfaces?  That is not out of the question?   How do you disambiguate?

Reply:
That's fine! In pr #644, we implement class factories for the ``geoips get command``. For example, if we had a plugin named ``Cloud-Height`` in two different interfaces, this is how it would work:
```bash
geoips get algorithm Cloud-Height
geoips get reader Cloud-Height
geoips get product-default Cloud-Height
...
```
You're forced to provide the interface name. This gives us enough information to find the correct plugin.

Reply:
It would look like this:
```
geoips list algorithms infrared
geoips list colormaps infrared
```

Reply:
I will say that ``list`` commands don't list specifically named plugins. If we were to do ``geoips list algorithms``, the only other arguments you could add are ``-p <package_name>``, ``--columns``, or ``--long``.

Reply:
mindy
  1:16 PM
I am playing with it a bit. I'm not sure how I feel with kind of lumping the "geoips list <interface_name>" and "geoips list <explicit_sub_cmd>" together under the same "list" command.  But I am leaning towards just merging it, and letting more people play with it, and getting feedback ?




NEW

[1:17](https://geoips.slack.com/archives/C0630KQ0KF0/p1721938643004979)
Because, at the same time, "geoips list interface algorithms" is also a "different" way of calling geoips list.  So either way is lumping 2 different types of calls to "geoips list" together, and I suppose fewer characters to type is as good of a reason to pick one over the other as any